chrome/content/zotero/xpcom/schema.js

```diff
@@ -3450,7 +3450,17 @@ Zotero.Schema = new function(){
 				}
 			}
 			
-			// TEMP: When adding 122, check whether IA.authorName fix in items.js::_loadAnnotations()
+			else if (i == 122) {
+				try {
+					await Zotero.DB.queryAsync("INSERT INTO fileTypes VALUES(8, 'ebook')");
+					await Zotero.DB.queryAsync("INSERT INTO fileTypeMIMETypes VALUES(8, 'application/epub+zip')");
+					// Incorrect, for compatibility
+					await Zotero.DB.queryAsync("INSERT INTO fileTypeMIMETypes VALUES(8, 'application/epub')");
+				}
+				catch (e) {}
```

Comment:
Oh, sorry, we still make the change to system.sql, and that's used on DB initialization. We just don't bump the system.sql version to trigger the old wipe-and-replace mechanism.

And this doesn't need to be in a try/catch — since we're setting userdata version 122 in this commit, where we'll also add the lines to system.sql, this won't run unless these rows don't exist.

Reply:
> And this doesn't need to be in a try/catch — since we're setting userdata version 122 in this commit, where we'll also add the lines to system.sql, this won't run unless these rows don't exist.

Oh, but I see why you had the try/catch — because an upgrade that does involve a system.sql bump still creates the system tables with the new system.sql, so the new rows exist. So I introduced an upgrade failure by removing this. Sorry about that.

But what we've done for past steps, and what we can do here, is just use REPLACE instead of INSERT. Fixed in 075457be4.